src/server.rs

```diff
@@ -8,7 +8,7 @@ use std::{
 };
 use tokio::sync::mpsc::*;
 
-pub fn initialize_server() -> Result<()> {
+pub fn initialize_server(alternate_config_dir: Option<std::path::PathBuf>) -> Result<()> {
```

Comment:
Maybe rename this to something like `config_dir_override`, so it's clearer that it takes precedence